src/coreclr/jit/codegenwasm.cpp

```diff
@@ -204,6 +528,162 @@ void CodeGen::genCodeForLclVar(GenTreeLclVar* tree)
+            break;
+
+        case PackOperAndType(GT_LE, TYP_INT):
+            ins = treeNode->IsUnsigned() ? INS_i32_le_u : INS_i32_le_s;
+            break;
+        case PackOperAndType(GT_LE, TYP_LONG):
+            ins = treeNode->IsUnsigned() ? INS_i64_le_u : INS_i64_le_s;
+            break;
+
+        case PackOperAndType(GT_GE, TYP_INT):
+            ins = treeNode->IsUnsigned() ? INS_i32_ge_u : INS_i32_ge_s;
+            break;
+        case PackOperAndType(GT_GE, TYP_LONG):
+            ins = treeNode->IsUnsigned() ? INS_i64_ge_u : INS_i64_ge_s;
+            break;
+
+        case PackOperAndType(GT_GT, TYP_INT):
+            ins = treeNode->IsUnsigned() ? INS_i32_gt_u : INS_i32_gt_s;
+            break;
+        case PackOperAndType(GT_GT, TYP_LONG):
+            ins = treeNode->IsUnsigned() ? INS_i64_gt_u : INS_i64_gt_s;
+            break;
+
+        default:
+            ins = INS_none;
+            NYI_WASM("genCodeForBinary");
+            break;
+    }
+
+    GetEmitter()->emitIns(ins);
+    genProduceReg(treeNode);
+}
+
+//------------------------------------------------------------------------
+// genCompareFloat: Generate code for comparing floats
+//
+// Arguments:
+//    treeNode - the compare tree
+//
+void CodeGen::genCompareFloat(GenTreeOp* treeNode)
+{
+    assert(treeNode->OperIsCmpCompare());
+
+    if ((treeNode->gtFlags & GTF_RELOP_NAN_UN) != 0)
+    {
+        NYI_WASM("genCompareFloat: unordered compares");
+    }
+
+    genConsumeOperands(treeNode);
+
+    instruction ins;
+    switch (PackOperAndType(treeNode->OperGet(), treeNode->gtOp1->TypeGet()))
+    {
+        case PackOperAndType(GT_EQ, TYP_FLOAT):
+            ins = INS_f32_eq;
+            break;
+        case PackOperAndType(GT_EQ, TYP_DOUBLE):
+            ins = INS_f64_eq;
+            break;
+        case PackOperAndType(GT_NE, TYP_FLOAT):
+            ins = INS_f32_ne;
+            break;
+        case PackOperAndType(GT_NE, TYP_DOUBLE):
+            ins = INS_f64_ne;
+            break;
+        case PackOperAndType(GT_LT, TYP_FLOAT):
+            ins = INS_f32_lt;
+            break;
+        case PackOperAndType(GT_LT, TYP_DOUBLE):
+            ins = INS_f64_lt;
+            break;
+        case PackOperAndType(GT_LE, TYP_FLOAT):
+            ins = INS_f32_le;
+            break;
+        case PackOperAndType(GT_LE, TYP_DOUBLE):
+            ins = INS_f64_le;
+            break;
+        case PackOperAndType(GT_GE, TYP_FLOAT):
+            ins = INS_f32_ge;
+            break;
+        case PackOperAndType(GT_GE, TYP_DOUBLE):
+            ins = INS_f64_ge;
+            break;
+        case PackOperAndType(GT_GT, TYP_FLOAT):
+            ins = INS_f32_gt;
+            break;
+        case PackOperAndType(GT_GT, TYP_DOUBLE):
+            ins = INS_f64_gt;
+            break;
+
+        default:
+            ins = INS_none;
+            NYI_WASM("genCodeForBinary");
+            break;
```

Comment:
```suggestion
            unreached();
```
(Not applicable if you do fold it into one function).